server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -718,7 +718,7 @@ protected void runInContext() {
                  getDynamicDataFromDB();
                  long interval = (Long) dbStats.get(uptime) - lastUptime;
                  long activity = (Long) dbStats.get(queries) - lastQueries;
-                 loadHistory.add(0, Double.valueOf(activity / interval));
+                 loadHistory.add(0, interval == 0 ? 0 : Double.valueOf(activity / interval));
                  int maxsize = DATABASE_SERVER_LOAD_HISTORY_RETENTION_NUMBER.value();
                  while (loadHistory.size() > maxsize) {
                      loadHistory.remove(maxsize - 1);
```

Comment:
additional bug @lucas-a-martins : it should delete any item beyond maxsize not maxsize -1
if you can take this in?
```suggestion
                     loadHistory.remove(maxsize);
```